Yoast/wordpress-seo
# Add an SKU assessment for product pages

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Implements the SKU assessment and temporarily disables it until the necessary data from Woo and Shopify become available.
* [yoastseo] Implements the SKU assessment and temporarily disables it until the necessary data from Woo and Shopify become available.
* [shopify-seo] Implements the SKU assessment and temporarily disables it until the necessary data from Woo and Shopify become available.

## Relevant technical choices:

* This is the first PR needed to introduce the SKU assessment in Woo and Shopify. The assessment is implemented in this PR but it's not working yet, as we don't yet access the data from the Woo and Shopify repositories.
This is why the applicability condition of the assessments is now set to false; it will be changed once the necessary data become available.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

#### Test the assessment with Woo settings
* Change the applicability condition in `ProductSKUAssessment.js` to `true` (line 64)
* Build and activate Yoast SEO Free 
* Require the current branch in the Woo `trunk`: `composer require yoast/wordpress-seo:dev-PC-89-add-an-sku-assessment@dev --dev` and Activate Yoast SEO Woo
* Open/create a product 
* Confirm that the SKU assessment appears in the SEO metabox
* Go the Inspect console
* Try and add some mocked data by typing the following script:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
    hasGlobalSKU: true,
    hasVariants: false,
  } );
  return data;
} );
```
* Add some text in the editor to make the assessment update
* Confirm that the assessment returns a green bullet and the feedback: `SKU: Good job!`
* Refresh the page
* Change the data to the following:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
  hasGlobalSKU: false,
hasVariants: true,
doAllVariantsHaveSKU: true,
  } );
  return data;
} );
```
* Confirm that you get a green bullet and the feedback: `SKU: Good job!`
* Turn on the cornerstone toggle and confirm that the assessment still appears with the same feedback


#### Test the assessment with Shopify settings
* Change the parameters values for the SKU assessment in `Assessor.js` from this:
```
new ProductSKUAssessment( {
			urlTitle: createAnchorOpeningTag( options.productIdentifierUrlTitle ),
			urlCallToAction: createAnchorOpeningTag( options.productIdentifierCTAUrl ),
			assessVariants: options.assessVariants,
		} ),
```
to this:
```
new ProductSKUAssessment( {
urlTitle: createAnchorOpeningTag( “https://yoa.st/4lx” ),
			urlCallToAction: createAnchorOpeningTag( “https://yoa.st/4lw” ),
			assessVariants: false,
} ),
```
* Build and activate Yoast SEO Free 
* Open/create a new product
* Go the Inspect console
* Try and add some mocked data by typing the following script:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
    hasGlobalSKU: true,
    hasVariants: false,
  } );
  return data;
} );
```
* Add some text in the editor to make the assessment update
* Confirm that the assessment returns a green bullet and the feedback: `SKU: Good job!`
* * Change the data to the following:
```
wp.hooks.addFilter( “yoast.analysis.data”, “wpseo-woocommerce-example-filter”, data => {
  data.customData = {};
  Object.assign( data.customData, {
  hasGlobalSKU: false,
hasVariants: true,
  } );
  return data;
} );
```
* Confirm that the assessment does not appear anymore.


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Install and activate Yoast SEO Free and Woo
* Open/create a new product
* Confirm that the SKU assessment is not showing in the SEO assessments section
* Install and activate Yoast SEO for Shopify app
* Open/create a new product
* Confirm that the SKU assessment is not showing in the SEO assessments section

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/PC-89
